src/container/cli.c

```diff
@@ -2535,9 +2550,31 @@ dc_cont_get_attr(tse_task_t *task)
 	for (i = 0, in->cagi_key_length = 0; i < args->n; i++)
 		in->cagi_key_length += strlen(args->names[i]) + 1;
 
-	rc = attr_bulk_create(args->n, (char **)args->names,
-			      (void **)args->values, (size_t *)args->sizes,
-			      daos_task2ctx(task), CRT_BULK_RW, &in->cagi_bulk);
+	/* no easy way to determine if a name storage address is likely
+	 * to cause an EFAULT during memory registration, so duplicate
+	 * name in heap
+	 */
+	D_ALLOC_ARRAY(new_names, args->n);
+	if (!new_names)
+		D_GOTO(out, rc = -DER_NOMEM);
+	for (i = 0 ; i < args->n ; i++) {
```

Comment:
This looks like the same for loop as L2550 so you could combine the two and calculate in->cagi_key_length in this loop, and change that strlen() to a strnlen(), as well as provide the correct size to the D_STRNDUP() on L2561

Reply:
right, will do. Thx!